meson.build

```diff
@@ -69,11 +77,33 @@ i18n.merge_file(
   install_dir: join_paths(datadir, 'applications'),
 )
 
-i18n.merge_file(
+# Validate the desktop file
+desktop_file_validate = find_program('desktop-file-validate', required:false)
+if desktop_file_validate.found()
+  test (
+    'Validate desktop file',
+    desktop_file_validate,
+    args: [ desktop_file.full_path() ],
+  )
+endif
+
+metainfo_file = i18n.merge_file(
   input: files('share/metainfo/net.lutris.Lutris.metainfo.xml'),
   output: 'net.lutris.Lutris.metainfo.xml',
   type: 'xml',
   po_dir: podir,
   install: true,
   install_dir: join_paths(datadir, 'metainfo'),
 )
+
+# Validate the metainfo file
+appstreamcli = find_program('appstream-util', required: false)
+if appstreamcli.found()
+  test (
+    'Validate metainfo file',
+    appstreamcli,
+    args: ['validate-relax', '--nonet', metainfo_file.full_path() ]
+  )
+endif
+
+meson.add_install_script('utils/meson_post_install.py')
```

Comment:
If you add a `meson_version: '>=0.57.0'` to the `project()` kwargs, you can use the gnome module's [`post_install()`](https://mesonbuild.com/Gnome-module.html#gnomepost_install) method to do this for you. It correctly handles different names for gtk-update-icon-cache, skips if destdir is set, calculates the right datadir etc...

Reply:
It only works well with >=0.59, given that Lutris stance of supporting distros from other eras I think this is the correct approach.

Reply:
Add a todo comment. :)

Also: it's difficult to know in a bubble what the project policy is, since there is no current minimum version specified. Nevertheless, it uses features not available in Debian Buster or Ubuntu 18.04.

Point taken on Meson 0.59, I overlooked that update-desktop-database is needed but wasn't in the initial implementation alongside update-icon-cache.

Reply:
> it uses features not available in Debian Buster or Ubuntu 18.04, dunno if that matters.

What features for example?

Reply:
I set a really low number as a test, and Meson reported the following feature warnings:

```
...

meson.build:29: WARNING: Project targeting '>=0.22.0' but tried to use feature introduced in '0.50.0': install arg in configure_file.
Configuring optional_settings.py using configuration
po/meson.build:6: WARNING: Project targeting '>=0.22.0' but tried to use feature introduced in '0.50.0': install_dir arg in i18n.gettext.

...

WARNING: Project specifies a minimum meson_version '>=0.22.0' but uses features which were added in newer versions:
 * 0.37.0: {'i18n.merge_file'}
 * 0.38.0: {'install_mode arg in install_data'}
 * 0.46.0: {'Python Module'}
 * 0.50.0: {'install arg in configure_file', 'install_dir arg in i18n.gettext'}
```

Reply:
Actually you could drop both 0.50 uses pretty easily.

Reply:
#4192

Reply:
Thanks, ill remove my commit setting it to 0.50 then